test/gerkin/features/steps/data_list.js

```diff
@@ -0,0 +1,97 @@
+const { spec, expect, stores } = require('pactum');
+const { When, Then, Given, After } = require('@cucumber/cucumber');
+
+const firstSearchedName = 'John';
+const secondSearchedName = 'Adrien';
+const baseUrl = 'http://localhost:3333/data/registry1/version1';
+const header = {
+  key: 'Information-Mediator-Client',
+  value: 'INSTANCE/CLASS/MEMBER/SUBSYSTEM',
+};
+
+// scenario 1
+Given('The user wants to search for "John Helmut" in the database', () => {
+  return firstSearchedName;
+});
+
+Given('The searched value exists in several records in the database', () => {
+  return 'Searched value does exist in the database';
+});
+
+When(
```

Comment:
Is it possible to not repeat requests in each segment? I would also make each section do only what's described. I'm not sure how this request library works but I would envision something like that:

(pseudo code, could be structured better)
```
let johnHelmutSearchRequest;
When(
  'The user triggers an action to search "John Helmut" in the database',
  async () => {
    johnHelmutSearchRequest = spec()
      .get(`${baseUrl}1?filter=FirstName&search=${firstSearchedName}`)
  }
);

When('The request with a valid payload is sent', () => {
  johnHelmutSearchRequest = johnHelmutSearchRequest.withHeaders(`${header.key}`, `${header.value}`)
});

Then(
  'The user receives a list with all records including "John Helmut"',
  async () => {
    await johnHelmutSearchRequest
      .expectStatus(200)
      .expectJsonLike({
        results: [
          {
            FirstName: 'John',
          },
        ],
      });
  }
);
```